src/rpm_ostree/cli_upgrade.rs

```diff
@@ -0,0 +1,23 @@
+//! Interface to `rpm-ostree upgrade --lock-finalization`.
+
+use super::Release;
+use failure::{bail, format_err, Fallible, ResultExt};
+
+/// Upgrade and leave the new deployment locked.
+pub fn locked_upgrade(release: Release) -> Fallible<Release> {
+    let cmd = std::process::Command::new("rpm-ostree")
+        .arg("upgrade")
```

Comment:
We just chatted about this OOB already, but just making a note here: this will have to change to `deploy` and the argument you give it would be `revision=$checksum`. Will fix up rpm-ostree.

Reply:
https://github.com/projectatomic/rpm-ostree/pull/1846